esp-hal/src/rng/ll.rs

```diff
@@ -0,0 +1,87 @@
+use crate::{clock::Clocks, peripherals::RNG, sync::Locked};
+
+// TODO: find a better place for these
+#[inline]
+#[cfg(soc_cpu_has_prv_mode)]
+fn tee_enabled() -> bool {
+    false
+}
+
+#[inline]
+fn current_cpu_cycles() -> usize {
+    cfg_if::cfg_if! {
+        if #[cfg(xtensa)] {
+            xtensa_lx::timer::get_cycle_count() as usize
+        } else if #[cfg(soc_cpu_has_csr_pc)] {
+            const PRV_M: usize = 3;
+            macro_rules! read_csr_fn {
```

Comment:
Can we use anything from https://docs.rs/riscv/latest/riscv/? Looks like the csr macros available there could replace this?